File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        pairs = {}
        for i in range(len(nums)):
            if nums[i] in pairs:
                return [pairs[nums[i]],i]
            else:
                pairs[target-nums[i]] = i

        

File: 0001-two-sum/0001-two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int,int> indexes = {};

    for(int i=0;i<nums.length;i++){
        if(indexes.containsKey(nums[i])){
            int? index = indexes[nums[i]];
            return [index!,i];
        }else{
            indexes[target-nums[i]] = i;
        }
    } 
    return [];   
  }
}